In [28]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/twitter-sentiment-analysis-preprocessed-dataset/y_test.csv
/kaggle/input/twitter-sentiment-analysis-preprocessed-dataset/X_test.csv
/kaggle/input/twitter-sentiment-analysis-preprocessed-dataset/y_train.csv
/kaggle/input/twitter-sentiment-analysis-preprocessed-dataset/X_train.csv
/kaggle/input/sentiments-deep-neural-network-data/y_train1.csv
/kaggle/input/sentiments-deep-neural-network-data/y_val1.csv
/kaggle/input/sentiments-deep-neural-network-data/y_test.csv
/kaggle/input/sentiments-deep-neural-network-data/X_test.csv
/kaggle/input/sentiments-deep-neural-network-data/y_train.csv
/kaggle/input/sentiments-deep-neural-network-data/X_train.csv
/kaggle/input/sentiments-deep-neural-network-data/X_val1.csv


# **Importing required dataset**

In [29]:
import pandas as pd 
import numpy as np
from keras.utils import to_categorical
from keras import models
from keras import layers

# **Importing datasets**

In [30]:
X_train1=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/X_train.csv")
X_test=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/X_test.csv")
y_train=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/y_train.csv")
y_test=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/y_test.csv")
y_val1=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/y_val1.csv")
X_val1=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/X_val1.csv")
y_train1=pd.read_csv("/kaggle/input/sentiments-deep-neural-network-data/y_train1.csv")
# X_train1=pd.read_csv("")

In [31]:
print(X_train.shape)

(48146, 1000)


# **Data Understanding**

In [32]:
X_train.shape

(48146, 1000)

In [33]:
X_train.head()

father  kid  run  thanks  cause  offer  bihday  model  love  time  ...  \
0       0    0    0       0      0      0       0      0     0     0  ...   
1       0    0    0       0      0      0       0      0     0     0  ...   
2       0    0    0       0      0      0       0      0     0     0  ...   
3       0    0    0       0      0      0       0      0     0     0  ...   
4       0    0    0       0      0      0       0      0     0     0  ...   

   nail  man  gif  fit  info  english  link  sound  trending  plus  
0     0    0    0    0     0        0     0      0         0     0  
1     0    0    0    0     0        0     0      0         0     0  
2     0    0    0    0     0        0     0      0         0     0  
3     0    0    0    0     0        0     0      0         0     0  
4     0    0    0    0     0        0     0      0         0     0  

[5 rows x 1000 columns]

In [34]:
X_test.shape

(5944, 1000)

In [35]:
y_train.shape

(53496, 1)

In [36]:
y_test.shape

(5944, 1)

In [37]:
type(y_test)

pandas.core.frame.DataFrame

In [38]:
y_test.head()

label
0      0
1      1
2      1
3      0
4      0

# **Building and Training the Neural Network**

**There are two types of models available in keras: the Model Sequential API and the Functional API.**

**For more complex deep learning models, you should use the Functional API because Sequential API does not allow you to create models that share layers or have multiple inputs or outputs.**

# **Sequential API model**

In [39]:
#**Set your model as the sequential model by running this command in a new cell:**
model=models.Sequential()

# **Creating layers for neural network**

**Add the input, hidden, and output layers. Between them, you will use dense on every layer and dropout to prevent overfitting.**

# **a] Input - Layer**

In [40]:
model.add(layers.Dense(50, activation = "relu", input_shape=(1000, )))

# **b] Hidden - Layers**

In [41]:
model.add(layers.Dropout(0.3,noise_shape=None,seed=None))
model.add(layers.Dense(50,activation="relu"))
model.add(layers.Dropout(0.2, noise_shape=None, seed=None))
model.add(layers.Dense(50, activation = "relu"))

# **c] Output - Layer**

**we will use the relu function within the hidden layers because it yields an acceptable result. relu stands for Rectified Linear Unit, and the function returns 0 if it receives any negative input or the value back for any positive value. Adding this function to your first layer removes the negative values by returning 0. In this tutorial, the relu function ensures all the values entering the input layer are positive values, which are necessary for the neural inputs.**

**At the output layer, you will use the sigmoid function, which maps the values 0 and 1. Since the output is positive (1) or negative (0), the sigmoid function will ensure the output layer produces an output that is either 0 or 1.**

In [42]:
model.add(layers.Dense(1, activation = "sigmoid"))
model.summary()

Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_4 (Dense)             (None, 50)                50050     
                                                                 
 dropout_2 (Dropout)         (None, 50)                0         
                                                                 
 dense_5 (Dense)             (None, 50)                2550      
                                                                 
 dropout_3 (Dropout)         (None, 50)                0         
                                                                 
 dense_6 (Dense)             (None, 50)                2550      
                                                                 
 dense_7 (Dense)             (None, 1)                 51        
                                                                 
Total params: 55,201
Trainable params: 55,201
Non-trai

# **Model compilation and configuration for training**

**The compile() function defines the model architecture. You implement the adam algorithm as your model’s optimizer in this definition. This algorithm is a gradient descent method based on approximating first-order and second-order moments. The metrics and loss parameters are very closely related. While the metric parameter defines how to judge your model’s performance, the loss parameter defines the quantity that the model seeks to minimize during training. The metric here is accuracy (the fraction of predictions the model gets correctly), while the loss is binary_crossentropy (a metric between the labels and predictions used when there are only two label classes [the positive 1 and the negative 0]).**


In [43]:
model.compile(optimizer="adam",loss="binary_crossentropy",metrics=["accuracy"])

# **Model training with neural network model**

**Now, you will be able to train your model. To do this, you will use a batch_size of 32 and just two epochs. The batch size is the number of samples that will be propagated through the neural network, and an epoch is an iteration over the entire training data. A larger batch size generally implies faster training but sometimes converges slower. Conversely, a smaller batch size is slower in training but can converge faster.**

**Fitting the model on the training data for training model**

In [53]:
results = model.fit(X_train1,y_train1,epochs= 5,batch_size =32,validation_data = (X_val1,y_val1))

Epoch 1/5
1505/1505 [==============================] - 5s 3ms/step - loss: 0.1372 - accuracy: 0.9421 - val_loss: 0.1263 - val_accuracy: 0.9488
Epoch 2/5
1505/1505 [==============================] - 5s 3ms/step - loss: 0.1191 - accuracy: 0.9488 - val_loss: 0.1232 - val_accuracy: 0.9493
Epoch 3/5
1505/1505 [==============================] - 5s 3ms/step - loss: 0.1097 - accuracy: 0.9514 - val_loss: 0.1302 - val_accuracy: 0.9503
Epoch 4/5
1505/1505 [==============================] - 5s 3ms/step - loss: 0.1065 - accuracy: 0.9528 - val_loss: 0.1151 - val_accuracy: 0.9544
Epoch 5/5
1505/1505 [==============================] - 5s 3ms/step - loss: 0.1017 - accuracy: 0.9544 - val_loss: 0.1140 - val_accuracy: 0.9546


In [59]:
y_train1.shape

(48146, 1)

# **Evaluating the Model**

# **1] Model evaluation with training data using accuracy_score**

In [54]:
scores=model.evaluate(X_train1,y_train1,verbose=0)
print("Accuracy: %.2f%%" % (scores[1]*100))

Accuracy: 95.91%


# **Observation:**

**Got 95.91% accuracy on training data using NN model**

# **2] Model evaluation with test data using accuracy_score**

In [55]:
scores=model.evaluate(X_test,y_test,verbose=0)
print("Accuracy: %.2f%%" % (scores[1]*100))

Accuracy: 95.26%


# **Accuracy_score for validation set**

In [56]:
scores=model.evaluate(X_val1,y_val1,verbose=0)
print("Accuracy: %.2f%%" % (scores[1]*100))

Accuracy: 95.46%


# **Observation:**

**Got 95.46% accuracy on test data using NN model**

In [61]:
from sklearn.metrics import classification_report,confusion_matrix

In [64]:
y_pred_train1= model.predict(X_train1)
y_pred_train1.shape

1505/1505 [==============================] - 2s 2ms/step


(48146, 1)

In [69]:
# print(confusion_matrix(y_train1,y_pred_train1))

ValueError: Classification metrics can't handle a mix of binary and continuous targets